src/models/ComponentAttributeDefinition.js

```diff
@@ -20,6 +20,9 @@ class ComponentAttributeDefinition {
    * @param {boolean} [props.required] - Attribute is required.
    * @param {ComponentAttributeDefinition[]} [props.definedAttributes] - Defined attributes for
    * this type.
+   * @param {ComponentAttributeDefinition[]} [props.itemDefinition] - Definition of the structure
+   * for individual array elements.
+   * @param {string} [props.itemType] - Type of individual array elements.
```

Comment:
Missing explanation on, type we can set on and if defined type is Object an itemDefinition must be provided

Reply:
done